iree/compiler/Codegen/Common/IREEComprehensiveBufferizePass.cpp

```diff
@@ -104,6 +104,10 @@ void IREEComprehensiveBufferizePass::runOnOperation() {
   options.testAnalysisOnly = testAnalysisOnly;
   options.printConflicts = printConflicts;
   options.alwaysAliasingWithDest = true;
+  // TODO(hanchung): Revisit if we want to disable the flag. memref.alloca ops
+  // are created when the flag is enabled. There are no alloca ops when the flag
+  // is disable. It looks like an upstream bug.
+  options.createDeallocs = false;
```

Comment:
This flag should not be activated. Otherwise, every buffer allocation will leak. The issue is caused by unfortunate layering in my code and a problem of the buffer deallocation pass with a HAL op.

This should fix this issue: https://reviews.llvm.org/D122304

Reply:
Thanks for the pointers! Cherry-picked the commit and reverted this change.

Reply:
cool cool, thanks the quick fix!